pkg/nodeagent/controller/healthcheck/reconciler.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/gardener/gardener/pkg/nodeagent/dbus"
+	"github.com/gardener/gardener/pkg/utils/flow"
+)
+
+// Reconciler checks for containerd and kubelet health and restarts them if required.
+type Reconciler struct {
+	Hostname                   string
+	Client                     client.Client
+	Recorder                   record.EventRecorder
+	DBus                       dbus.DBus
+	HealthCheckers             []HealthChecker
+	HealthCheckIntervalSeconds int32
+}
+
+// Reconcile executes all defined healtchecks
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+	log.Info("Reconcile Healthchecks")
+	node := &metav1.PartialObjectMetadata{}
+	node.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Node"))
+
+	if err := r.Client.Get(ctx, request.NamespacedName, node); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	taskFns := []flow.TaskFn{}
+	for _, hc := range r.HealthCheckers {
+		taskFns = append(taskFns, hc.Check)
+	}
+
+	err := flow.Parallel(taskFns...)(ctx)
+
+	return reconcile.Result{RequeueAfter: time.Duration(r.HealthCheckIntervalSeconds) * time.Second}, err
```

Comment:
```suggestion
	return reconcile.Result{RequeueAfter: time.Duration(r.HealthCheckIntervalSeconds) * time.Second}, flow.Parallel(taskFns...)(ctx)
```

Reply:
I solved it differently, because when `Reconcile` returns an non-empty response and a non-nil error, it spams the logs with messages like this
```
Dec 07 21:24:01 machine-shoot--local--local-local-bfc8c-szg6h gardener-node-agent[48218]: {"level":"info","ts":"2023-12-07T21:24:01.344Z","msg":"Warning: Reconciler returned both a non-zero result and a non-nil error. The result will always be ignored if the error is non-nil and the non-nil error causes reqeueuing with exponential backoff. For more details, see: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler","controller":"healthcheck","controllerGroup":"","controllerKind":"Node","Node":{"name":"machine-shoot--local--local-local-bfc8c-szg6h"},"namespace":"","name":"machine-shoot--local--local-local-bfc8c-szg6h","reconcileID":"c287f301-564d-419a-92d8-ae97033a8dd9"}
```